samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,108 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("\n");
+
+  while (true) {
```

Comment:
If there's an option to check if the controller is still connected, you should use it (even if it might still not work) to drop out of this loop / inform the user that the input no longer works and the app will exit in X seconds.

Reply:
I can't find such a function. `SDL_IsGameController()` is probably the closest I can find - good enough?

Reply:
I fail to see how SDL_GameControllerEventState could be used in a helpful manner here. Would you please enlighten me on the details?

Reply:
It can't, ignore what I said, my fingers were faster than my eyes.

Reply:
I have proposed `SDL_GameControllerGetAttached` in another (non-outdated) discussion.